src/helpers.rs

```diff
@@ -456,6 +456,17 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    /// Dispatches to appropriate implementations for reading an OsString from Memory,
+    /// depending on the interpretation target.
+    fn read_os_str_from_target_str(&self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
```

Comment:
If you want, this would be a nice opportunity to learn how to use `Cow`. You can make the type of this function
```rust
fn read_os_str_from_c_str<'a>(&'a self, scalar: Scalar<Tag>) -> InterpResult<'tcx, Cow<'a, OsStr>>
    where
        'tcx: 'a,
        'mir: 'a,
```
and then in the linux/macos case do `map(Cow::Borrowed)` and in the other case `.map(Cow::Owned)`. That way, we can avoid unnecessary copies.

Reply:
We can also do the Cow thing later though as you suggested. In that case, please leave a comment like `FIXME: Use Cow to avoid copies`.